### PR TITLE
MODCLUSTER-794 mod_proxy_cluster gets stuck with 20 nodes

### PR DESCRIPTION
**Pull request by jajik**
Fix and  new test for https://issues.redhat.com/browse/MODCLUSTER-794.

~~Add test for https://issues.redhat.com/browse/MODCLUSTER-794.~~

~~This branch is based on #120 so I'm opening this as a draft until it is merged.~~

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]
Please rebase.

**Comment by jajik** [comment]
Rebase done. Because the issue has not been fixed yet, I disabled this test as well (in CI).

**Comment by rhusar** [comment]
So this is a test case for [MODCLUSTER-794](https://issues.redhat.com/browse/MODCLUSTER-794) – I personally prefer to get test cases merged as disabled and the person fixing the issue fixes and enables the test – @jfclere can weigh in.

**Comment by jajik** [comment]
> So this is a test case for [MODCLUSTER-794](https://issues.redhat.com/browse/MODCLUSTER-794)

Yes.

**Comment by rhusar** [comment]
Converted to draft PR since there are 'wip' commits.

**Comment by jajik** [comment]
Marking ready for review. It's no longer the test only, but there is a fix too.

**Comment by rhusar** [comment]
> Marking ready for review. It's no longer the test only, but there is a fix too.

Great. @jfclere please review.

**Review by jfclere — APPROVED** [review]

**Review by rhusar — APPROVED** [review]
LGTM now.

**Comment by rhusar** [comment]
Thnaks @jfclere @jajik

File: test/MODCLUSTER-794/testit.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+. includes/common.sh
+
+# first stop any previously running tests.
+tomcat_all_stop
+tomcat_all_remove
+httpd_all_clean
+
+MPC_NAME=MODCLUSTER-794 MPC_CONF=MODCLUSTER-794/mod_proxy_cluster.conf httpd_run
+
+for i in {1..20}; do
```
Comment (rhusar) [inline test/MODCLUSTER-794/testit.sh:21]: Oh, almost forgot to mention, do we really need to boot 20 tomcats here @jajik ? Why not go with lower number like "3" and boot appropriate number of tomcats?

Comment (jajik) [reply test/MODCLUSTER-794/testit.sh:21]: I wanted to preserve the defaults, and resource-wise it didn't seem like a big issue. Of course, we can change it eventually.

**Pull request merged**